models/nft/ethereum/nft_ethereum_top_sales.sql

```diff
@@ -0,0 +1,82 @@
+{{ config(
+     alias = 'top_sales',
+     materialized='incremental',
+     file_format = 'delta',
+     incremental_strategy = 'merge',
+     unique_key = ['rn'],
+     post_hook='{{ expose_spells(\'["ethereum"]\',
+                                 "sector",
+                                 "nft",
+                                 \'["Henrystats"]\') }}'
+     )
+ }}
+
+{% if not is_incremental() %}
+WITH src AS
+(
+    SELECT
+        nft_contract_address
+        , token_id as nft_token_id
+        , seller
+        , amount_original as price
+        , tx_hash
+        , ROW_NUMBER() OVER (ORDER BY amount_original DESC) as rn
```

Comment:
alright ser @henrystats -- i fixed the logic here to make this work in a timely manner. there's a big if/assumption here. based on our convo in comments, if this is true:
the output should be top-50 sales of all time, at any given time. the output will have 50 rows at all times. updates will only happen if a rank has shifted.

with that in mind, we don't need to partition by `nft_contract_address` in the row number function. if we do, then we will obtain top-50 sales of every single nft contract address, therefore output would be count of nft contracts * 50 -- leading to an exponentially higher number. this is why it was taking much longer previously.

is this as expected?

Reply:
if we do need top 50 for each address, then we will need to just take out all "is incremental" logic and do a full read each time, change from merge to delete/insert, ideally that'd be quicker. easiest thing is to just take out incremental altogether if this scenario is what is needed, so it'd just replace the table each time.

Reply:
Thank you very much ser @jeff-dude. 

The top 50 is for the top sales for all the nft_contract_address, it was why I felt an incremental logic would help since the table would only look at trades in the past 7 days and check for trades higher than the lowest value in the previous top 50. 

If we have to look through the entire table, then would it be better to just query this from dune.com? 

cc @0xRobin (you pointed originally that without incremental then it'd be better to have these on dune.com so I'm curious what you think ser. 

Also tagging @0xBoxer for your thoughts on this.

Reply:
the simple approach is materialize as a table. it'll overwrite each time. the initial build on incremental wasn't taking long anyway. we can see how this runs.

Reply:
```
select
  nft_contract_address,
  count(1)
from
  test_schema.git_7c4244a7_nft_ethereum_top_sales
group by
  nft_contract_address
order by
  2 desc
```

Reply:
alrighty ser, thank you very much!

Reply:
not sure if it was related to this spell, but dbt won't let us do insert-overwrite strategy for incremental on delta table format. it also won't work over sql endpoints, which is what our infrastructure uses currently. this is the only way, as we don't have unique columns to join on for a merge, it'll create cross joins that explode the row counts.